docs/docs/life-and-times-of-a-gatsby-build.md

```diff
@@ -2,7 +2,305 @@
+
+The `gatsby develop` command doesn't perform some of the build steps that the `gatsby build` command does, but starts up a development server that you can use to preview your site in the browser -- like a runtime. When you run `gatsby build` (build time) there won't be a browser available, so your site needs to be capable of protecting calls to browser based APIs.
+
+To gain a greater understanding of what happens when you run either command, it can be helpful to look at the information that Gatsby is reporting back and break it down.
+
+### Understanding gatsby develop (runtime)
+
+Using `gatsby develop` runs a server in the background enabling useful features like hot reloading and Gatsby’s data explorer.
+
+Gatsby develop is optimized for rapid feedback and extra debugging information. The output of running `gatsby develop` in a fresh install of the Gatsby default starter looks like this:
+
+```shell
+success open and validate gatsby-configs - 0.051 s
+success load plugins - 0.591 s
+success onPreInit - 0.015 s
+success initialize cache - 0.019 s
+success copy gatsby files - 0.076 s
+success onPreBootstrap - 0.021 s
+success source and transform nodes - 0.082 s
+success Add explicit types - 0.018 s
+success Add inferred types - 0.106 s
+success Processing types - 0.080 s
+success building schema - 0.266 s
+success createPages - 0.014 s
+success createPagesStatefully - 0.067 s
+success onPreExtractQueries - 0.017 s
+success update schema - 0.034 s
+success extract queries from components - 0.222 s
+success write out requires - 0.044 s
+success write out redirect data - 0.014 s
+success Build manifest and related icons - 0.110 s
+success onPostBootstrap - 0.130 s
+⠀
+info bootstrap finished - 3.674 s
+⠀
+success run static queries - 0.057 s — 3/3 89.08 queries/second
+success run page queries - 0.033 s — 5/5 347.81 queries/second
+success start webpack server - 1.707 s — 1/1 6.06 pages/second
+```
+
+You can maybe take guesses at what some of the steps are doing, but others require context of Gatsby internals to make sense of (don't worry, the rest of this guide will give more explanations).
+
+### Understanding gatsby build (build time)
+
+Gatsby build is made for when you’ve added the finishing touches to your site and everything looks great. `gatsby build` creates a version of your site with optimizations like packaging up your site’s config, data, and code, and creating all the HTML that eventually gets [rehydrated](/docs/glossary#hydration) into a React app.
+
```

Comment:
maybe add it here:

```suggestion
You can run and test the output from `gatsby build` with the `gatsby serve` command. 

```

maybe additional: Compared to the `gatsby develop` the `gatsby serve` only rely on the build, so if you update the code you need rerun the `gatsby build`. 

or maybe this can be an own section: "Differences between `gatsby develop` and `gatsby serve`: